specutils/spectra/spectral_coordinate.py

```diff
@@ -0,0 +1,651 @@
+        ----------
+        value
+
+        Notes
+        -----
+        More information on the equations dictating the transformations can be
+        found in the astropy documentation [1]_.
+
+        References
+        ----------
+        .. [1] Astropy documentation: https://docs.astropy.org/en/stable/units/equivalencies.html#spectral-doppler-equivalencies
+
+        """
+        if value is not None and value not in DOPPLER_CONVENTIONS:
+            raise ValueError("Unrecognized velocity convention: {}.".format(
+                value))
+
+        if self._doppler_conversion.convention is not None:
+            raise ValueError("Doppler convention has already been set. Use "
+                             "the `to` method to update the stored value.")
+
+        self._doppler_conversion = self._doppler_conversion._replace(convention=value)
+
+    @property
+    def radial_velocity(self):
+        """
+        Radial velocity of target relative to the observer.
+
+        Returns
+        -------
+        `u.Quantity`
+            Radial velocity of target.
+
+        Notes
+        -----
+        This is different from the ``.radial_velocity`` property of a
+        coordinate frame in that this calculates the radial velocity with
+        respect to the *observer*, not the origin of the frame.
+        """
+        return self._calculate_radial_velocity(self._observer, self._target)
+
+    @property
+    def redshift(self):
+        """
+        Redshift of target relative to observer. Calculated from the radial
+        velocity.
+
+        Returns
+        -------
+        float
+            Redshift of target.
+        """
+        return self.radial_velocity.to('', equivalencies=RV_RS_EQUIV)
+
+    @staticmethod
+    def _calculate_radial_velocity(observer, target):
+        """
+        Compute the line-of-sight velocity from the observer to the target.
+
+        Parameters
+        ----------
+        observer : `BaseCoordinateFrame`
+            The frame of the observer.
+        target : `BaseCoordinateFrame`
+            The frame of the target.
+
+        Returns
+        -------
+        `Quantity`
+            The radial velocity of the target with respect to the observer.
+        """
+        # Convert observer and target to ICRS to avoid finite
+        # differencing calculations that lack numerical precision.
+        if observer is None:
+            raise ValueError("No observer has been specified; cannot "
+                             "calculate radial velocity.")
+
+        observer_icrs = observer.transform_to(ICRS)
+
+        if target is None:
+            raise ValueError("No target has been specified; cannot calculate "
+                             "radial velocity.")
+
+        target_icrs = target.transform_to(ICRS)
+
+        d_pos = (target_icrs.data.without_differentials() -
+                 observer_icrs.data.without_differentials()).to_cartesian()
+
+        pos_hat = d_pos / (d_pos.norm() or 1)
```

Comment:
This isn't right because there's no guarantee norm() yields a scalar - `d_pos` can be a complicated vector-y thing.

Also it yields a confusing warning when I just display this because numpy says this kind of usage is deprecated anyway.  I think instead you want something like this:
```suggestion
        dpnorm = d_pos.norm()
        # reset any that are 0 to 1 to avoid nans from 0/0.
        dpnorm.ravel()[dpnorm.ravel() == 0] = 1*dpnorm.unit

        pos_hat = d_pos / dpnorm
```